tests/functional/adapter/persist_docs/test_persist_docs.py

```diff
@@ -0,0 +1,145 @@
+import pytest
+from dbt.tests.util import run_dbt, check_relations_equal, run_sql_with_adapter
+
+from tests.functional.adapter.persist_docs.fixtures import (
+    seed_csv,
+    table_model,
+    view_model,
+    incremental_model,
+    table_profile_yml,
+    view_profile_yml,
+    incremental_profile_yml
+)
+
+
+@pytest.mark.iceberg
+class TestPersistDocsBase:
+    """
+    Testing persist_docs functionality
+    """
+
+    @property
+    def schema(self):
+        return "default"
+
+    # everything that goes in the "seeds" directory
+    @pytest.fixture(scope="class")
+    def seeds(self):
+        return {
+            "seed.csv": seed_csv,
+        }
+
+
+class TestPersistDocsTable(TestPersistDocsBase):
+
+    @pytest.fixture(scope="class")
+    def project_config_update(self):
+        return {
+            "name": "persist_docs_tests",
+            "models": {
+                "+persist_docs": {
+                    "relation": True,
+                    "columns": True
+                }
+            },
+            "seeds": {
+                "+column_types": {
+                    "date": "timestamp(6)"
+                }
+            }
+        }
+
+    # everything that goes in the "models" directory
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {
+            "table_model.sql": table_model,
+            "table_persist_docs.yml": table_profile_yml,
+        }
+
+    def test_run_seed_test(self, project):
+        # seed seeds
+        results = run_dbt(["seed"], expect_pass=True)
+        assert len(results) == 1
+        results = run_dbt(["run"], expect_pass=True)
+        assert len(results) == 1
+        # test tests
+        results = run_dbt(["test"], expect_pass=True)
+        assert len(results) == 3
+
+
+# Comments on views are not supported in Trino engine
+# https://github.com/trinodb/trino/issues/10705
+class TestPersistDocsView(TestPersistDocsBase):
+
+    @pytest.fixture(scope="class")
+    def project_config_update(self):
+        return {
+            "name": "persist_docs_tests",
+            "models": {
+                "+persist_docs": {
+                    "relation": True
```

Comment:
we basically test whether dbt is persisting docs for relations and columns, but can we check whether the comments were set as expected on the underlying database?

Reply:
Thanks for a valid comment. This test should be handled by macro https://github.com/starburstdata/dbt-trino/blob/master/dbt/include/trino/macros/catalog.sql which creates a file `target/catalog.json` generated by `dbt docs generate` command. 

In the `get_catalog` macro table and column comments are returned as `null`. Information schema tables in Trino miss comment columns. I will create a follow-up ticket in Trino for it.  

Docs are generated by a file located in `target/manifest.json` and all comments are present in dbt-docs and persisted in the Trino.

Reply:
https://github.com/trinodb/trino/issues/13705